### PR TITLE
BUG: GetGlobalDefaultSplitter to a global singleton

### PR DESCRIPTION
**Pull request by thewtex**
Ensure a shared instance across binaries via itk::Singleton.

### CODE REVIEW

**Review by blowekamp — APPROVED** [review]
LGTM

**Review by dzenanz — APPROVED** [review]

File: Modules/Core/Common/src/itkImageSourceCommon.cxx [hunk 1]
```diff
@@ -15,27 +15,29 @@
...
-namespace
+struct ImageSourceCommonGlobals
 {
-std::once_flag                   globalDefaultSplitterOnceFlag;
-ImageRegionSplitterBase::Pointer globalDefaultSplitter;
-} // namespace
+  ImageSourceCommonGlobals() { m_GlobalDefaultSplitter = ImageRegionSplitterSlowDimension::New().GetPointer(); };
+
+  ImageRegionSplitterBase::Pointer m_GlobalDefaultSplitter;
+};
```
Comment (N-Dekker) [inline Modules/Core/Common/src/itkImageSourceCommon.cxx:31]: Please consider using the C++ Rule of Zero: remove the user-defined default-constructor, and initialize m_GlobalDefaultSplitter at its declaration instead:

    struct ImageSourceCommonGlobals
    {
        ImageRegionSplitterBase::Pointer m_GlobalDefaultSplitter{ ImageRegionSplitterSlowDimension::New().GetPointer() };
    };

Comment (thewtex) [reply Modules/Core/Common/src/itkImageSourceCommon.cxx:31]: @N-Dekker thanks for the review and tip. I tried this, but a number of the tests segfault -- the initialization order is important here.

Comment (N-Dekker) [reply Modules/Core/Common/src/itkImageSourceCommon.cxx:31]: @thewtex Thanks for considering my suggestion to use the Rule of Zero. I don't really understand how such a code simplification could cause segfaults, do you? Of do you think those segfaults just happen coincidentally?

When having the proposed simplification, the initialization still takes place during the default-construction of the `ImageSourceCommonGlobals` struct. So as far as I can see, there is no reordering of the initialization.

**Pull request merged**